### ISSUE TITLE
Blocks: Differences between editor and frontend

### ISSUE THREAD

**Issue by scruffian** [WordPress/gutenberg#29976]
## Latest Comments:
|Editor|Frontend|
| --- | --- |
|<img width="1000" alt="Screenshot 2021-03-18 at 11 45 49" src="https://user-images.githubusercontent.com/275961/111621207-9dcf5700-87df-11eb-8da1-4cd68a6e2767.png">|<img width="889" alt="Screenshot 2021-03-18 at 11 45 44" src="https://user-images.githubusercontent.com/275961/111621200-99a33980-87df-11eb-8e99-330cf800ed4a.png">|
|The comment text is smaller in the editor than in the frontend|

## Archives:
|Editor|Frontend|
| --- | --- |
|<img width="470" alt="Screenshot 2021-03-18 at 11 50 37" src="https://user-images.githubusercontent.com/275961/111621762-47164d00-87e0-11eb-840d-442f6eba2f96.png">|<img width="430" alt="Screenshot 2021-03-18 at 11 50 41" src="https://user-images.githubusercontent.com/275961/111621793-4d0c2e00-87e0-11eb-9caa-50c71085a0cc.png">|
|The dropdown appearance is different and has a label|

## File block:
|Editor|Frontend|
| --- | --- |
|<img width="376" alt="Screenshot 2021-03-18 at 11 35 25" src="https://user-images.githubusercontent.com/275961/111619940-09b0c000-87de-11eb-99fb-b8d519f4371d.png">|<img width="307" alt="Screenshot 2021-03-18 at 11 35 44" src="https://user-images.githubusercontent.com/275961/111619984-159c8200-87de-11eb-8c52-0b0591ccee6d.png">|
|The file name appears like a link in the frontend but not in the editor|

## Search block:
|Editor|Frontend|
| --- | --- |
|<img width="1437" alt="Screenshot 2021-03-18 at 11 54 44" src="https://user-images.githubusercontent.com/275961/111622214-ce63c080-87e0-11eb-8764-1b9ceb40f402.png">|<img width="886" alt="Screenshot 2021-03-18 at 11 54 49" src="https://user-images.githubusercontent.com/275961/111622228-d3287480-87e0-11eb-92e5-bdf1fa2a3891.png">|
|The placeholder and label font is a different font family and color.|

## Cover block:
|Editor|Frontend|
| --- | --- |
|<img width="1160" alt="Screenshot 2021-03-18 at 12 10 32" src="https://user-images.githubusercontent.com/275961/111623957-f9e7aa80-87e2-11eb-8bf1-3009c533279e.png">|<img width="891" alt="Screenshot 2021-03-18 at 12 10 38" src="https://user-images.githubusercontent.com/275961/111623975-fe13c800-87e2-11eb-9dbe-e26f39e49223.png">|
|The cover block has a bckground color in the editor but not the frontend|

## Quotation:
|Editor|Frontend|
| --- | --- |
|<img width="344" alt="Screenshot 2021-03-18 at 12 13 53" src="https://user-images.githubusercontent.com/275961/111624356-7c706a00-87e3-11eb-99f8-f1e0afc17032.png">|<img width="383" alt="Screenshot 2021-03-18 at 12 13 47" src="https://user-images.githubusercontent.com/275961/111624346-78dce300-87e3-11eb-92bc-9b460069f21f.png">|
|The "This citation is bold" is a different font size|

When the block styles are disabled:

|Editor|Frontend|
| --- | --- |
|<img width="643" alt="Screenshot 2021-03-19 at 14 44 43" src="https://user-images.githubusercontent.com/275961/111798195-bdd74700-88c1-11eb-9936-3c032c3ef4ef.png">|<img width="681" alt="Screenshot 2021-03-19 at 14 44 49" src="https://user-images.githubusercontent.com/275961/111798203-c0d23780-88c1-11eb-889e-16a6cbfbe44d.png">|
|In the frontend the citation text is italic, in the editor its smaller|

## Paragraph:
|Editor|Frontend|
| --- | --- |
|<img width="793" alt="Screenshot 2021-03-18 at 12 16 23" src="https://user-images.githubusercontent.com/275961/111624671-e1c45b00-87e3-11eb-9a4d-f7158f9cca83.png">|<img width="850" alt="Screenshot 2021-03-18 at 12 16 30" src="https://user-images.githubusercontent.com/275961/111624642-db35e380-87e3-11eb-9399-833bbf7df8cc.png">|
|<img width="869" alt="Screenshot 2021-03-19 at 13 00 45" src="https://user-images.githubusercontent.com/275961/111784713-a09b7c00-88b3-11eb-8480-fc5c4642c37e.png">|<img width="949" alt="Screenshot 2021-03-19 at 13 00 51" src="https://user-images.githubusercontent.com/275961/111784748-aabd7a80-88b3-11eb-9038-bb2f12858466.png">|
|There is additional space in the text around the drop cap. Also when paragraphs have a background color the edge of the block aligns with the text in the editor, but the text aligns with other text blocks in the frontend.|

## Columns:
|Editor|Frontend|
| --- | --- |
| <img width="880" alt="Screenshot 2021-03-18 at 13 36 40" src="https://user-images.githubusercontent.com/275961/111635413-5d77d500-87ef-11eb-9b73-1e01395ee149.png">|<img width="881" alt="Screenshot 2021-03-18 at 13 36 48" src="https://user-images.githubusercontent.com/275961/111635435-61a3f280-87ef-11eb-9921-48c2b4a12f57.png">|
|There is a bigger margin at the top of the columns block in the editor.|

## Post Title:
|Editor|Frontend|
| --- | --- |
|<img width="513" alt="Screenshot 2021-03-19 at 13 05 09" src="https://user-images.githubusercontent.com/275961/111784981-f112d980-88b3-11eb-96b2-98833350f494.png">|<img width="295" alt="Screenshot 2021-03-19 at 13 05 20" src="https://user-images.githubusercontent.com/275961/111785000-f6702400-88b3-11eb-8f6d-977a9377f721.png">|
|The font size is bigger in the editor|

Opting out of block styles:

## Table:
|Editor|Frontend|
| --- | --- |
|<img width="841" alt="Screenshot 2021-03-19 at 14 11 47" src="https://user-images.githubusercontent.com/275961/111793415-28d24f00-88bd-11eb-8fa6-7ea61c655036.png">|<img width="716" alt="Screenshot 2021-03-19 at 14 11 38" src="https://user-images.githubusercontent.com/275961/111793434-2cfe6c80-88bd-11eb-817e-96a68c567fec.png">|
|Tables in the frontend don't have a border|

## Buttons:
|Editor|Frontend|
| --- | --- |
|<img width="897" alt="Screenshot 2021-03-19 at 14 31 29" src="https://user-images.githubusercontent.com/275961/111796266-ec542280-88bf-11eb-8c28-76aa0c40cc39.png">|<img width="970" alt="Screenshot 2021-03-19 at 14 31 22" src="https://user-images.githubusercontent.com/275961/111796283-f0804000-88bf-11eb-875c-4f843945a613.png">|
|<img width="911" alt="Screenshot 2021-03-19 at 14 31 44" src="https://user-images.githubusercontent.com/275961/111796334-fd049880-88bf-11eb-812c-e84d6c3d93b8.png">|<img width="926" alt="Screenshot 2021-03-19 at 14 31 51" src="https://user-images.githubusercontent.com/275961/111796351-00981f80-88c0-11eb-96d7-cae22bc16ff7.png">|
|In the frontend buttons overlap each other horizontally. The large button at the bottom is aligned left in the editor but centered in the editor.|

## Group:
|Editor|Frontend|
| --- | --- |
|<img width="1440" alt="Screenshot 2021-03-19 at 14 37 58" src="https://user-images.githubusercontent.com/275961/111797154-caa76b00-88c0-11eb-990b-adf4bfb8c0af.png">|<img width="1440" alt="Screenshot 2021-03-19 at 14 37 42" src="https://user-images.githubusercontent.com/275961/111797164-cf6c1f00-88c0-11eb-903a-a96e90a502d2.png">|
|The group block extends to the edge of the screen in the editor but not in the frontend|

## Embed:
|Editor|Frontend|
| --- | --- |
|<img width="860" alt="Screenshot 2021-03-19 at 14 53 49" src="https://user-images.githubusercontent.com/275961/111799873-53270b00-88c3-11eb-8f77-5e2de7b0d519.png">|<img width="899" alt="Screenshot 2021-03-19 at 14 53 56" src="https://user-images.githubusercontent.com/275961/111799954-6508ae00-88c3-11eb-83c9-c2fad85ab063.png">|
|<img width="869" alt="Screenshot 2021-03-19 at 14 53 20" src="https://user-images.githubusercontent.com/275961/111799887-55896500-88c3-11eb-9bf2-0da5d8efe1eb.png">|<img width="849" alt="Screenshot 2021-03-19 at 14 53 26" src="https://user-images.githubusercontent.com/275961/111799969-676b0800-88c3-11eb-988b-149339dc631a.png">|
|<img width="853" alt="Screenshot 2021-03-19 at 14 53 05" src="https://user-images.githubusercontent.com/275961/111799893-57532880-88c3-11eb-8b1d-49225011dce6.png">|<img width="814" alt="Screenshot 2021-03-19 at 14 53 10" src="https://user-images.githubusercontent.com/275961/111799972-689c3500-88c3-11eb-8120-c4d824189f46.png">|
|<img width="898" alt="Screenshot 2021-03-19 at 14 52 48" src="https://user-images.githubusercontent.com/275961/111799894-57532880-88c3-11eb-83e6-76a12d0fb5c1.png">|<img width="842" alt="Screenshot 2021-03-19 at 14 52 54" src="https://user-images.githubusercontent.com/275961/111799975-6934cb80-88c3-11eb-8cc6-74c44d37a73b.png">|
|<img width="828" alt="Screenshot 2021-03-19 at 14 52 28" src="https://user-images.githubusercontent.com/275961/111799897-58845580-88c3-11eb-9254-18b26b7e458b.png">|<img width="766" alt="Screenshot 2021-03-19 at 14 52 33" src="https://user-images.githubusercontent.com/275961/111799979-69cd6200-88c3-11eb-99b1-1f7c2c4f2c55.png">|
|<img width="855" alt="Screenshot 2021-03-19 at 14 52 05" src="https://user-images.githubusercontent.com/275961/111799906-591cec00-88c3-11eb-914a-527f7ef485f6.png">|<img width="840" alt="Screenshot 2021-03-19 at 14 52 09" src="https://user-images.githubusercontent.com/275961/111799981-6a65f880-88c3-11eb-82a3-0fa11c5ade3b.png">|
|<img width="892" alt="Screenshot 2021-03-19 at 14 51 35" src="https://user-images.githubusercontent.com/275961/111799914-59b58280-88c3-11eb-8f2c-5a928dedf4e9.png">|<img width="754" alt="Screenshot 2021-03-19 at 14 51 46" src="https://user-images.githubusercontent.com/275961/111799984-6afe8f00-88c3-11eb-8991-86b9571e926d.png">|
|The embed blocks are a different size on the frontend and the editor|

## Separator:
|Editor|Frontend|
| --- | --- |
|With block styles:|
|<img width="919" alt="Screenshot 2021-03-23 at 16 54 05" src="https://user-images.githubusercontent.com/275961/112186071-b0d79200-8bf8-11eb-97ac-f41684bd23a2.png">|<img width="962" alt="Screenshot 2021-03-23 at 16 54 11" src="https://user-images.githubusercontent.com/275961/112186078-b3d28280-8bf8-11eb-9e38-559e4803101a.png">|
|Without block styles|
|<img width="1440" alt="Screenshot 2021-03-23 at 16 55 22" src="https://user-images.githubusercontent.com/275961/112186089-b7660980-8bf8-11eb-89b4-e8cc03ffdddc.png">|<img width="1440" alt="Screenshot 2021-03-23 at 16 55 33" src="https://user-images.githubusercontent.com/275961/112186099-b92fcd00-8bf8-11eb-85e6-c590ddf1a5b8.png">|
|The 100px max with is really problematic. Also not sure why the color gets lost|

There is also a difference in margins for all blocks

TODO:
- [x] Latest Comments #30134
- [x] Archives #30141 
- [x] File block #30148
- [x] Search block #30176
- [x] Cover block 
- [x] Quote #30190
- [x] Paragraph
- [x] Columns #30229
- [x] Post Title
- [x] Table
- [x] Buttons #30253
- [x] Group
- [x] Embed #30254 
- [x] Separator #30255

**Comment by youknowriad** [comment]
What's the problem for the buttons block (aside the margins between the blocks which are a separate thing cross blocks)

**Comment by scruffian** [comment]
> What's the problem for the buttons block (aside the margins between the blocks which are a separate thing cross blocks)

That's all :)

**Comment by youknowriad** [comment]
@scruffian Do you think we can add a TODO list to the issue description to help clarify what's done, what's remaining...

**Comment by scruffian** [comment]
> @scruffian Do you think we can add a TODO list to the issue description to help clarify what's done, what's remaining...

Done! I added another one I found for background colors...

**Comment by youknowriad** [comment]
@scruffian do you know we can clarify what is different for each block like:

 - Buttons: margin
 - Block: color for links, background ...

Sorry if I'm asking too much 😬 It's something you know right away cause you did the screenshot but it might not be obvious by just looking at the screenshots.

**Comment by MaggieCabrera** [comment]
I suppose you did these with Empty theme, right? I just wanted to point out that even though empty theme doesn't add css, it does include some [custom supports](https://github.com/WordPress/theme-experiments/blob/master/emptytheme/functions.php) and as we've found, removing things like `add_theme_support( 'editor-styles' );` causes differences between the frontend and the editor.

**Comment by scruffian** [comment]
I added some details about the differences between the blocks under each screenshot.

**Comment by scruffian** [comment]
> I suppose you did these with Empty theme, right? I just wanted to point out that even though empty theme doesn't add css, it does include some custom supports and as we've found, removing things like add_theme_support( 'editor-styles' ); causes differences between the frontend and the editor.

Yes, I did these tests with block styles enabled (`add_theme_support( 'wp-block-styles' )`). We should also do the same tests without this enabled.

These styles should not be loading in the editor unless the theme opts in since https://github.com/WordPress/gutenberg/pull/29252 merged.

**Comment by scruffian** [comment]
I added some more differences which I found when they block styles are disabled.

**Comment by youknowriad** [comment]
@scruffian I haven't been able to reproduce the "Cover" block issue above, any example markup?

**Comment by MaggieCabrera** [comment]
> @scruffian I haven't been able to reproduce the "Cover" block issue above, any example markup?

I couldn't reproduce myself, and I think he was using the content from https://theamdemo.wordpress.com/2019/06/24/gutenberg-cover/ I can give you access to it

**Comment by scruffian** [comment]
This seems to happen when using a background color that isn't defined, e.g.

```
<!-- wp:cover {"url":"https://theamdemo.files.wordpress.com/2020/10/qi-bin-w4hbafegiac-unsplash.jpg","id":28,"overlayColor":"red","minHeight":100,"minHeightUnit":"vh"} -->
<div class="wp-block-cover has-red-background-color has-background-dim" style="min-height:100vh"><img class="wp-block-cover__image-background wp-image-28" alt="" src="https://theamdemo.files.wordpress.com/2020/10/qi-bin-w4hbafegiac-unsplash.jpg" data-object-fit="cover"/><div class="wp-block-cover__inner-container"><!-- wp:paragraph {"align":"center","placeholder":"Write title…","fontSize":"large"} -->
<p class="has-text-align-center has-large-font-size">Full height Cover</p>
<!-- /wp:paragraph --></div></div>
<!-- /wp:cover -->
```

**Comment by youknowriad** [comment]
For the drop cap, for me, it's the same between frontend and backend, you can compare properly by having a long text in the paragraph to wrap the drop cap cause otherwise the difference you'll have is related to the margins (which is a cross block issue I'm leaving for its own issue as it's kind of big)

**Comment by youknowriad** [comment]
The Post Title block seems to be in the same size for me (default heading sizes)

**Comment by youknowriad** [comment]
Tables also look the same for me, I do see borders on the frontend.

**Comment by youknowriad** [comment]
I have a PR for the buttons block but I wasn't able to reproduce the center aligned one.

**Comment by youknowriad** [comment]
I guess the group block difference above is about the "padding" in the site editor right? If that's the case, I'll leave that separate because this padding has some wider implications

**Comment by paaljoachim** [comment]
Checking in.
What is the status of this issue?

**Comment by scruffian** [comment]
All of these issues were fixed I think we can close this.

**Comment by scott8035** [comment]
Was there a common theme among the fixes to all these issues? I'm having the same problem with a custom block created with ACF Pro, and I've no idea where to start looking.

**Comment by scruffian** [comment]
@scott8035, it will really depend on the block. Ideally you want the markup to be the same in the editor and on the frontend, and then also to use the same CSS in both places.
